dspace-api/src/main/java/org/dspace/content/enhancer/impl/MapConverterValueEnhancer.java

```diff
@@ -0,0 +1,129 @@
+import org.dspace.content.Item;
+import org.dspace.content.MetadataField;
+import org.dspace.content.enhancer.AbstractItemEnhancer;
+import org.dspace.content.enhancer.ItemEnhancer;
+import org.dspace.content.service.ItemService;
+import org.dspace.content.service.MetadataFieldService;
+import org.dspace.core.Context;
+import org.dspace.core.exception.SQLRuntimeException;
+import org.dspace.services.ConfigurationService;
+import org.dspace.util.SimpleMapConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Implementation of {@link ItemEnhancer} that add metadata values on the given
+ * item from the result of some mapConverter
+ * Only consider 1 to 1 value enhancement and no additional security/authority/language settings form the origin value.
+ * e.g. dc.type (dspace) -> dc.type (coar)
+ *
+ * @author Florian Gantner (florian.gantner@uni-bamberg.de)
+ *
+ */
+public class MapConverterValueEnhancer extends AbstractItemEnhancer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MapConverterValueEnhancer.class);
+
+    @Autowired
+    private ItemService itemService;
+
+	@Autowired
+	private MetadataFieldService metadatafieldService;
+
+	@Autowired
+	private ConfigurationService configurationService;
+
+    private String sourceEntityType;
+
+    private String sourceItemMetadataField;
+    
+    private String targetItemMetadataField;
+
+	private boolean useDefaultLanguage;
+
+	private SimpleMapConverter converter;
+
+    @Override
+    public boolean canEnhance(Context context, Item item) {
+        return sourceEntityType == null || sourceEntityType.equals(itemService.getEntityType(item));
+    }
+
+    @Override
+    public void enhance(Context context, Item item) {
+        try {
+		if (StringUtils.isBlank(sourceItemMetadataField) || Objects.isNull(converter) || StringUtils.isBlank(targetItemMetadataField)) return;
+			String sourceval, targetval, calculatedval;
+			sourceval = itemService.getMetadata(item, sourceItemMetadataField);
+			targetval = itemService.getMetadata(item, targetItemMetadataField);
+			if (StringUtils.isNotBlank(sourceval)) {
+				calculatedval = converter.getValue(sourceval);
+				if (StringUtils.isNotBlank(targetval) && !targetval.contentEquals(calculatedval)) {
+					// replace mdv if it's different
+					removeTargetMetadata(context, item);
+					addTargetMetadata(context, item, calculatedval);
+				} else if (StringUtils.isBlank(targetval)) {
+					// set new value
+					addTargetMetadata(context, item, calculatedval);
+				}
+			} else if (StringUtils.isBlank(sourceval) && StringUtils.isNotBlank(targetval)) {
+				// remove value
+				removeTargetMetadata(context, item);
+			}
+        } catch (SQLException e) {
+            LOGGER.error("An error occurs enhancing item with id {}: {}", item.getID(), e.getMessage(), e);
+            throw new SQLRuntimeException(e);
```

Comment:
This leads to a situation where the execution of subsequent enhancers is skipped since `ItemEnhancerServiceImpl.enhance(c, i)` does not handle exceptions of type `SQLRuntimeException`.

Reply:
The Exception was adopted from the existing RelatedEntityItemEnhancer. Nevertheless I've added some more general logging of the Exception in this Enhancer